node/setup.go

```diff
@@ -83,6 +83,12 @@ func initDBs(
 	closers := []closer{}
 	blockStore := store.NewBlockStore(blockStoreDB)
 	closers = append(closers, blockStoreDB.Close)
+	defer func() {
+		if rerr != nil {
+			blockStore.Close()
+			_ = closeFn()
+		}
+	}()
```

Comment:
Isn't the function calling this responsible for also closing the resources if there's an error (and not this function)